torch/csrc/jit/codegen/cuda/mma_utils.cpp

```diff
@@ -0,0 +1,371 @@
+
+#include <torch/csrc/jit/codegen/cuda/expr_evaluator.h>
+#include <torch/csrc/jit/codegen/cuda/ir_printer.h>
+#include <torch/csrc/jit/codegen/cuda/mma_utils.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+namespace mma_util {
+
+namespace {
+
+// Utility for mma dimension matching
+enum class MmaDimension { M = 0, N, K };
+
+// Utility for mma dimension matching, assumes the innermost
+//  3 dimensions are the mma operand dimensions, i.e. mnk.
```

Comment:
I think we eventually want this to be generic for tensor contractions, we can leave it as is for now, but I don't think we want simple assumptions about layout coming into mma, we should make the interface explicit as we may actually get something that doesn't fall precisely into MNK where there's multiple and potentially non-contiguous dimensions that participate in them.

Reply:
Yes we will have a generic version in a separate PR. The generic matching logic seems to be a bit complex to handle in the same PR especially for tensor contractions. 

This part is only used for validating that the correct axes are being swizzled with mma pattern so that mma gives expected result. They have to be the axes that logically represent the m-n-k axes that actually participate in the matmuls. For gemm cases it's very simple. For generalized cases the `m,n,k` could each be actually multiple axes, and additional matching rules might be needed here.